### PR TITLE
Prometheus

### PR DESCRIPTION
**Pull request by rakita**

### CODE REVIEW

File: parity/export_hardcoded_sync.rs [hunk 1]
```diff
@@ -0,0 +1,123 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
```
Comment (adria0) [inline parity/export_hardcoded_sync.rs:1]: Is this related to Prometheus?

File: parity/configuration.rs [hunk 2]
```diff
@@ -159,13 +166,14 @@ impl Configuration {
...
         let format = self.format()?;
+        let metrics_conf = self.metrics_config()?;
         let keys_iterations = NonZeroU32::new(self.args.arg_keys_iterations)
             .ok_or_else(|| "--keys-iterations must be non-zero")?;
 
         let cmd = if self.args.flag_version {
             Cmd::Version
         } else if self.args.cmd_signer {
-            let authfile = ::signer::codes_path(&ws_conf.signer_path);
+            let authfile = crate::signer::codes_path(&ws_conf.signer_path);
```
Comment (adria0) [inline parity/configuration.rs:176]: Is this related to Prometheus?

File: parity/rpc.rs [hunk 3]
```diff
@@ -180,7 +182,7 @@ pub fn new_ws<D: rpc_apis::Dependencies>(
     let signer_path;
     let path = match conf.support_token_api {
         true => {
-            signer_path = ::signer::codes_path(&conf.signer_path);
+            signer_path = crate::signer::codes_path(&conf.signer_path);
```
Comment (adria0) [inline parity/rpc.rs:185]: Is this related to Prometheus?

File: parity/export_hardcoded_sync.rs (hunk 1 above)
Comment (rakita) [reply parity/export_hardcoded_sync.rs]: Dead code left after merge. Will be removed.

File: parity/configuration.rs (hunk 2 above)
Comment (rakita) [reply parity/configuration.rs:176]: The answer is no, but it is just syntax change needed fo the rust 2018 edition.

File: parity/rpc.rs (hunk 3 above)
Comment (rakita) [reply parity/rpc.rs:185]: The answer is no, but it is just syntax change needed fo the rust 2018 edition.

File: parity/configuration.rs (hunk 2 above)
Comment (adria0) [reply parity/configuration.rs:176]: Oh, yes, I do not know that I was thinking... Lol

File: scripts/prometheus/docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      dockerfile: scripts/docker/alpine/Dockerfile
+      context: ../..
+    ports:
+      - '30303:30303'
+      - '30303:30303/udp'
+      - '8545:8545'
+    links:
+      - prometheus
+
+    entrypoint: ["/home/openethereum/openethereum","--metrics","--metrics-interface=all"]
```
Comment (adria0) [inline scripts/prometheus/docker-compose.yaml:15]:
```suggestion
    entrypoint: ["/home/parity/parity","--metrics","--metrics-interface=all"]
```

File: parity/rpc.rs (hunk 3 above)
Comment (rakita) [reply parity/rpc.rs:185]: This is still valid with 2015

File: Cargo.toml [hunk 5]
```diff
@@ -4,6 +4,7 @@ name = "parity-ethereum"
 # NOTE Make sure to update util/version/Cargo.toml as well
 version = "2.5.13"
 license = "GPL-3.0"
+edition = "2018"
```
[the remark this answers is not in the document]
Comment (rakita) [reply Cargo.toml:7]: Using edition 2018 is step forward and this does not change the version of the compiler we are using. I would like this to stay, but for the current state, I don't have a strong opinion about it.

File: util/runtime/Cargo.toml [hunk 6]
```diff
@@ -7,5 +7,7 @@ version = "0.1.0"
 authors = ["Parity Technologies <admin@parity.io>"]
 
 [dependencies]
-futures = "0.1"
-tokio = "0.1.22"
+futures03 = {package = "futures",  version = "0.3.5", default-features = false, features = ["compat"] }
+futures = { package = "futures", version = "0.1" }
+tokio-compat = "0.1.6"
+tokio = { version = "0.1.22"}
```
[the remark this answers is not in the document]
Comment (rakita) [reply util/runtime/Cargo.toml:13]: I already tested this and I feel like it is a waste to just remove it, cause we could use it in the future.

File: util/journaldb/src/overlayrecentdb.rs [hunk 7]
```diff
@@ -288,15 +288,19 @@ impl JournalDB for OverlayRecentDB {
...
 
-    fn mem_used(&self) -> usize {
-        let mut mem = self.transaction_overlay.mem_used();
-        let overlay = self.journal_overlay.read();
-
-        mem += overlay.backing_overlay.mem_used();
-        mem += overlay.pending_overlay.heap_size_of_children();
-        mem += overlay.journal.heap_size_of_children();
+    fn get_sizes(&self, sizes: &mut BTreeMap<String, usize>) {
+        //sizes.insert(String::from("db_overlay_recent_transactions_size"), refs_size); //TODO MemoryDB
```
Comment (adria0) [inline util/journaldb/src/overlayrecentdb.rs:292]: Could you descibe with more detail what is this TODO, please?

Comment (rakita) [reply util/journaldb/src/overlayrecentdb.rs:292]: The current version of MemoryDB is just returning 0 on `mem_used()` and we don't have access to the underlying HashMap that is used in MemoryDB to at least get `len()`. The first intention for TODO is to change this when we update MemoryDB to new version, but now when I am thinking about it, we could just copy memory-db into project and add `lenI()` function.

**Review by adria0 — APPROVED** [review]
